common/src/main/java/com/wynntils/features/EmeraldPouchHotkeyFeature.java

```diff
@@ -0,0 +1,197 @@
+
+                pouchSwitch:
+                switch (emeraldPouches.size()) {
+                    default -> {
+                        Integer slotNumber = -1;
+                        ItemStack slotStack = null;
+
+                        boolean alreadyHasNonEmpty = false;
+                        Integer usedPouch = -1;
+                        ItemStack pouchStack = null;
+                        for (Integer key : emeraldPouches.keySet()) {
+                            if (emeraldPouches.get(key).b.a > 0
+                                    && !alreadyHasNonEmpty) { // Found one pouch with a nonzero balance, remember this
+                                alreadyHasNonEmpty = true;
+                                usedPouch = key; // Save our pouch slot ID
+                                pouchStack = emeraldPouches.get(key).a;
+                            } else if (emeraldPouches.get(key).b.a
+                                    > 0) { // Another pouch has a non-zero balance; notify user
+                                McUtils.sendMessageToClient(
+                                        new TranslatableComponent("feature.wynntils.emeraldPouchKeybind.multipleFilled")
+                                                .withStyle(ChatFormatting.DARK_RED));
+                                break pouchSwitch;
+                            }
+                        }
+
+                        // At this point, we have either multiple pouches with no emeralds, or multiple pouches but
+                        // only one with a non-zero balance
+                        // Check to make sure we don't have a bunch of zero balances - if we do, open largest capacity
+                        if (!alreadyHasNonEmpty) {
+                            Map.Entry<Integer, Pair<ItemStack, Pair<Integer, Integer>>> largest = null;
+                            for (Map.Entry<Integer, Pair<ItemStack, Pair<Integer, Integer>>> entry :
+                                    emeraldPouches.entrySet()) {
+                                if (largest == null || entry.getValue().b.b > largest.getValue().b.b) {
+                                    largest = entry;
+                                }
+                            }
+                            slotNumber = largest.getKey();
+                            slotStack = largest.getValue().a;
+                        }
+
+                        // Now, we know we have 1 used and 1+ empty pouches - open the used one we saved from before
+                        if (slotNumber == -1) {
+                            slotNumber = usedPouch;
+                        }
+                        if (slotStack == null) {
+                            slotStack = pouchStack;
+                        }
+                        if (slotNumber < 9) {
+                            // sendPacket uses raw slot numbers, we need to remap the hotbar
+                            slotNumber += 36;
```

Comment:
I may have said this already before, but this is rather weird and suggests a better alternative way of accessing slot numbers